sopel/modules/instagram.py

```diff
@@ -20,17 +21,27 @@
     from json import loads
 
 
+LOGGER = logging.getLogger(__name__)
 INSTAGRAM_REGEX = r'(https?:\/\/(?:www\.){0,1}instagram\.com\/([a-zA-Z0-9_\.]{,30}\/)?p\/[a-zA-Z0-9_-]+)'
 
-
 # TODO: Parse Instagram profile page
 
 
+class ParseError(Exception):
+    pass
+
+
 @module.url(INSTAGRAM_REGEX)
 def instaparse(bot, trigger, match):
+    instagram_url = match.group(1)
     # Get the embedded JSON
-    json = get_insta_json(match.group(1))
-    bot.say(parse_insta_json(json))
+    json = get_insta_json(instagram_url)
+    try:
+        bot.say(parse_insta_json(json))
+    except ParseError:
+        LOGGER.exception(
+            "Unable to find instagram's payload for URL %s", instagram_url)
+        bot.say('Unable to parse this instagram URL. :(')
```

Comment:
```suggestion
        bot.say('Unable to parse this Instagram URL. :( It's probably a temporary error; try again in a bit.')
```

Ain't gonna try and pep8-wrap text in GitHub's suggestion editor, but yeah. Aside from capitalizing "Instagram" I think it's reasonably important that whoever posted the URL doesn't feel like a goof, nor think the bot is dumb for being unable to parse something that "wOrKs In My BrOwSeR".